src/semgrep_agent/findings.py

```diff
@@ -99,9 +116,22 @@ def to_dict(self) -> Mapping[str, Any]:
 
 @dataclass(frozen=True)
 class FindingSets:
-    baseline: Set[Finding] = field(default_factory=set)
-    current: Set[Finding] = field(default_factory=set)
+    baseline_map: Dict[FindingKey, List[Finding]] = field(default_factory=dict)
+    current_map: Dict[FindingKey, List[Finding]] = field(default_factory=dict)
+
+    def expensive_new(self) -> Set[Finding]:
+        return self.current_set() - self.baseline_set()
```

Comment:
The index-based deduplication here is going to be yielding some confusing output. See this diff:


```diff
- eval("pre-vetted, safe, string")
+ eval(request.GET.foo)
  print("here comes the dangerous code, bail before running it in prod")
  if not app.DEBUG:
      return
  eval(request.GET.foo)
```

The semgrep-agent would now output line 5 as the new finding, right? Cause its index changed from 0 to 1, and index 1 is the never-before seen index. And a human code reviewer might dismiss that as a false positive looking at the debug mode gate.

So perhaps we'll need to look at the git diff to figure out what to report as new. Should we just ship this version now anyway since it's a bit better than what we have today?

Reply:
Short answer is that I don't see a good technical solution to this problem without really understanding the code that's written, as distinguishing between a line move and a delete + add is nigh impossible.

So I'd go with just leaving this as a wart on the solution.

Reply:
If we allow this limitation, wouldn't this same thing be accomplished with the few-line change of making the original `findings.current.update(Finding.from_semgrep_result(...))` line we have be locally aware of the `index` of the result, and then including the index in the hash? I'm thinking this, in pseudocode:

```python
finding = Finding.from_semgrep_result(...)
while finding in findings.current:
    finding.index += 1
findings.current.add(finding)
```